checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -772,6 +772,12 @@ private void addAnnotationForAnd(
             return;
         }
 
+        AnnotatedTypeMirror leftType = getAnnotatedType(leftExpr);
+        if (leftType.hasAnnotation(NN) || leftType.hasAnnotation(POS)) {
```

Comment:
Oh right.  Thanks!  @kelloggm, would you please add the code in my comment as a test case?